community/kernel/src/main/java/org/neo4j/kernel/impl/newapi/AllStoreHolder.java

```diff
@@ -74,16 +74,19 @@
     private final StoreReadLayer read;
     private final Lazy<ExplicitIndexTransactionState> explicitIndexes;
 
-    AllStoreHolder( RecordStorageEngine engine, Supplier<ExplicitIndexTransactionState> explicitIndexes )
+    AllStoreHolder( StorageEngine engine,
+                    StorageStatement statement,
+                    Supplier<ExplicitIndexTransactionState> explicitIndexes )
     {
-        read = engine.storeReadLayer();
-        statement = read.newStatement();
-        NeoStores stores = engine.testAccessNeoStores();
+        this.read = engine.storeReadLayer();
+        this.statement = statement; // use provided statement, to assert no leakage
+        this.explicitIndexes = Suppliers.lazySingleton( explicitIndexes );
+
+        StoreHolder stores = engine.stores();
```

Comment:
The `StorageEngine` should ideally not expose direct access to the underlying stores, but instead hide them behind interfaces such as `StoreReadLayer`, and eventually `org.neo4j.internal.kernel.api.Read` I suppose.